workflow/executor/resource.go

```diff
@@ -114,8 +118,27 @@ func (g gjsonLabels) Get(label string) string {
 	return gjson.GetBytes(g.json, label).String()
 }
 
+func (we *WorkflowExecutor) signalMonitoring() {
+	log.Infof("Starting signal monitoring")
+	sigs := make(chan os.Signal, 1)
```

Comment:
This looks very similar to `monitorAnnotations`. I think it could do with more commentary to explain how it works and what it is for.